src/VideoClip.js

```diff
@@ -34,17 +33,21 @@ class VideoClip extends MC.BrowserClip {
     video.muted = true;
     const canvas = this.context.getElements("canvas")[0];
     const ctx = canvas.getContext("2d");
-    const scaleX = this.width / video.videoWidth;
-    const scaleY = this.width / video.videoWidth;
-    video.addEventListener(
-      "loadedmetadata",
-      () => {
-        canvas.style.transform = `scale(${scaleX}, ${scaleY})`;
-        canvas.width = video.videoWidth;
-        canvas.height = video.videoHeight;
-      },
-      false
-    );
+
+    const loadedmetadataListener = () => {
+      const scaleX = this.width / video.videoWidth;
+      const scaleY = this.width / video.videoWidth;
+      canvas.style.transform = `scale(${scaleX}, ${scaleY})`;
+      canvas.width = video.videoWidth;
+      canvas.height = video.videoHeight;
+      video.removeEventListener(
+        "loadedmetadata",
+        loadedmetadataListener,
+        false
+      );
+    };
+
+    video.addEventListener("loadedmetadata", loadedmetadataListener, false);
```

Comment:
Considering you want this to run once, couldn't we set the option `once` to true so it will remove it after execution?